core/contracts/financial-templates/implementation/PricelessPositionManager.sol

```diff
@@ -320,7 +326,9 @@ contract PricelessPositionManager is FeePayer, AdministrateeInterface {
         } else {
             // Decrease the sponsors position size of collateral and tokens.
             _removeCollateral(positionData.rawCollateral, collateralRedeemed);
-            positionData.tokensOutstanding = positionData.tokensOutstanding.sub(numTokens);
+            FixedPoint.Unsigned memory newTokenCount = positionData.tokensOutstanding.sub(numTokens);
+            require(newTokenCount.isGreaterThanOrEqual(minSponsorTokens));
```

Comment:
+1 This is good because you can still redeem all your tokens (like 324) and if you redeem any less then there is a lower bound on how much you can leave behind.